examples/ShoppingAds/GetProductBiddingCategoryConstant.php

```diff
@@ -0,0 +1,179 @@
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        // Creates the query.
+        $query = 'SELECT product_bidding_category_constant.localized_name,
+            product_bidding_category_constant.product_bidding_category_constant_parent
+            FROM product_bidding_category_constant
+            WHERE product_bidding_category_constant.country_code IN (\'US\')
+            ORDER BY product_bidding_category_constant.localized_name ASC';
+
+        // Creates a list of top level category nodes.
+        $rootCategories = [];
+        // Creates a map of category ID to category node for all categories found in the results.
+        // This Map is a convenience lookup to enable fast retrieval of existing nodes.
+        $biddingCategories = [];
+
+        // Performs the search request.
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        // Iterates over all rows in all pages to extract the result.
+        foreach ($response->iterateAllElements() as $googleAdsRow) {
+            /**
+             * @var GoogleAdsRow $googleAdsRow
+             * @var ProductBiddingCategoryConstant $productBiddingCategory
+             */
+            $productBiddingCategory = $googleAdsRow->getProductBiddingCategoryConstant();
+            $localizedName = $productBiddingCategory->getLocalizedNameUnwrapped();
+            $resourceName = $productBiddingCategory->getResourceName();
+            if (!array_key_exists($resourceName, $biddingCategories)) {
+                $biddingCategories[$resourceName] = [];
+            }
+
+            // Ensures that the name attribute for the node is set. Name will be null for nodes added
+            // to biddingCategories as a result of being a parentNode below.
+            $biddingCategories[$resourceName]['localizedName'] = $localizedName;
+
+            if ($productBiddingCategory->getProductBiddingCategoryConstantParent() != null) {
+                $parentResourceName =
+                    $productBiddingCategory->getProductBiddingCategoryConstantParentUnwrapped();
+                if (!array_key_exists($parentResourceName, $biddingCategories)) {
+                    $biddingCategories[$parentResourceName] = [];
+                }
+                $biddingCategories[$parentResourceName]['children'][$resourceName] = $biddingCategories[$resourceName];
+            } else {
+                $rootCategories[$resourceName] = $biddingCategories[$resourceName];
```

Comment:
Since arrays in PHP aren't passed by reference, if a child bidding category is encountered after its parent bidding category is assigned to the rootCategories array, would that child be reflected in the final rootCategories array?

Reply:
Good one, tx